backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+	}
+
+	for _, cstatus := range pod.Status.ContainerStatuses {
+		if cstatus.Name == "main" {
+			if cstatus.LastTerminationState.Terminated == nil {
+				return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+			} else {
+				if cstatus.LastTerminationState.Terminated.ExitCode == 0 {
+					job.Status.State = lmevalservicev1beta1.CompleteJobState
+					job.Status.Reason = lmevalservicev1beta1.SucceedReason
+				} else {
+					job.Status.State = lmevalservicev1beta1.CompleteJobState
+					job.Status.Reason = lmevalservicev1beta1.FailedReason
+					job.Status.Message = cstatus.LastTerminationState.Terminated.Reason
+
+				}
+				if err := r.Status().Update(ctx, job); err != nil {
+					log.Error(err, "unable to update EvalJob status", "state", job.Status.State)
+					return ctrl.Result{}, err
+				}
+				r.Recorder.Event(job, "Normal", "PodCompleted",
+					fmt.Sprintf("The pod for the EvalJob %s in namespace %s has completed",
+						job.Name,
+						job.Namespace))
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+}
+
+func (r *EvalJobReconciler) getPod(ctx context.Context, job *lmevalservicev1beta1.EvalJob) (*corev1.Pod, error) {
+	var pod = corev1.Pod{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: job.Namespace, Name: job.Name}, &pod); err != nil {
+		return nil, err
+	}
+	for _, ref := range pod.OwnerReferences {
+		if ref.APIVersion == job.APIVersion &&
+			ref.Kind == job.Kind &&
+			ref.Name == job.Name {
+
+			return &pod, nil
+		}
+	}
+	return nil, fmt.Errorf("pod doesn't have proper entry in the OwnerReferences")
+}
+
+func (r *EvalJobReconciler) deleteJobPod(ctx context.Context, job *lmevalservicev1beta1.EvalJob) error {
+	pod := corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      job.Status.PodName,
+			Namespace: job.Namespace,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: job.APIVersion,
+					Kind:       job.Kind,
+					Name:       job.Name,
+				},
+			},
+		},
+	}
+	return r.Delete(ctx, &pod, &client.DeleteOptions{})
+}
+
+func (r *EvalJobReconciler) handleComplete(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	if job.Status.CompleteTime == nil {
+		r.Recorder.Event(job, "Normal", "JobCompleted",
+			fmt.Sprintf("Tthe EvalJob %s in namespace %s has completed",
+				job.Name,
+				job.Namespace))
+		// TODO: final wrap up/clean up
+		current := v1.Now()
+		job.Status.CompleteTime = &current
+		if err := r.Status().Update(ctx, job); err != nil {
+			log.Error(err, "failed to update status for completion")
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *EvalJobReconciler) handleCancel(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	// delete the pod and update the state to complete
```

Comment:
I think the same concern here around log retention would be at play. A user may want to retain the logs from the pod even after cancellation so that they can analyze the progress/problems from the job.

Reply:
got it. we can do that in the driver directly.